src/test/java/guitests/PanelsTest.java

```diff
@@ -1,80 +1,63 @@
 package guitests;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertTrue;
-import static org.junit.Assert.fail;
+import static ui.components.KeyboardShortcuts.*;
 
-import ui.TestController;
-import ui.issuepanel.FilterPanel;
-import ui.issuepanel.PanelControl;
-import static ui.components.KeyboardShortcuts.CREATE_RIGHT_PANEL;
-import static ui.components.KeyboardShortcuts.MAXIMIZE_WINDOW;
-import static ui.components.KeyboardShortcuts.SWAP_PANEL_LEFT;
+import java.util.concurrent.atomic.AtomicBoolean;
 
 import org.junit.Test;
-import org.loadui.testfx.exceptions.NoNodesFoundException;
+import org.loadui.testfx.utils.TestUtils;
 
-import javafx.scene.control.Label;
-import javafx.scene.text.Text;
+import javafx.scene.Node;
 import javafx.scene.input.KeyCode;
+import javafx.scene.text.Text;
+import ui.TestController;
 import ui.UI;
-import util.PlatformEx;
+import ui.issuepanel.AbstractPanel;
+import ui.issuepanel.FilterPanel;
+import ui.issuepanel.PanelControl;
 import util.events.PanelClickedEventHandler;
 
 public class PanelsTest extends UITest {
     
-    private static class Bool {
-        public boolean value = false;
-        public void negate() {
-            value = !value;
-        }
-    }
-
-    // TODO check if interactions result in any effects
     @Test
     public void panelsTest() {
         PanelControl panels = TestController.getUI().getPanelControl();
 
-        Bool eventTriggered = new Bool();
+        final AtomicBoolean eventTriggered = new AtomicBoolean(false);
 
-        UI.events.registerEvent((PanelClickedEventHandler) e -> eventTriggered.negate());
+        UI.events.registerEvent((PanelClickedEventHandler) e -> eventTriggered.set(true));
 
         pushKeys(MAXIMIZE_WINDOW);
-
         pushKeys(CREATE_RIGHT_PANEL);
+        waitUntilNodeAppears(((FilterPanel) panels.getPanel(0)).getFilterTextField());
+
         type("repo:dummy2/dummy2");
         push(KeyCode.ENTER);
 
         // Click
-        eventTriggered.value = false;
+        eventTriggered.set(false);
         Text name1 = ((FilterPanel) panels.getPanel(1)).getNameText();
         click(name1);
-        assertTrue(eventTriggered.value);
+        TestUtils.awaitCondition(eventTriggered::get);
 
-        // Drag
-        // TODO find a way to actually reorder the panels, and make it compatible with swap key command
+        // Reorder panels by dragging, and wait for them to swap position
+        final AbstractPanel firstPanel = panels.getPanel(0);
+        Node from = ((FilterPanel) panels.getPanel(1)).getCloseButton();
+        Node to = ((FilterPanel) panels.getPanel(0)).getFilterTextField();
```

Comment:
Why don't we use close button as well here? Might be confusing in the future. If there is a reason why we use filter text field instead of close button, maybe put a comment about it?